File: delete_sub_str.cc
// delete_sub_str.cpp : Defines the entry point for the console application.
#include"stdio.h"
#include"stdlib.h"
#include<iostream>
using namespace std;
int delete_sub_str(const char *str, const char *sub_str, char *result_str)
{
  int count=0;
  int i=0,j=0;
  while ( str[i]!=sub_str[j]) i++;
  for ( int k=0;k<i;k++)
  {
    *result_str++=str[k];
  }

  while ( '\0' != str[i])
  {
    while ( str[i++]==sub_str[j++] && '\0' != sub_str[j]);
    if (str[i]!='\0')
      count++;
    j=0;
    while ( str[i] != sub_str[j] && '\0' != str[i])
    {
      *result_str++=str[i];
      i++;
    }

  }
  *result_str='\0';
  return count;
}

int delete_sub_str(char str[], const char *sub_str)
{
  int count=0;
  bool found=true;
  int i=0,j=0,p=0,nav=0;
  while(found)
  {
    i=0;
    j=0;
    found=false;
    while ( str[i]!=sub_str[j]) i++;
    p=i;

    while (sub_str[j] && str[i] && str[i++]==sub_str[j++]);
    if (!sub_str[j])
    {
      found=true;
      count++;
    }
    else
    {
      return count;
    }

    nav=i;
    while (str[nav])
    {
      str[p++]=str[nav++];
    }
    if (str[nav]=='\0')
      str[p]='\0';
  }

  return count;
}

int count_sub(char *source,char *sub)
{
  int result=0;
  int i=0,j=0;
  while (source[i])
  {
    j=0;
    while ( source[i++]!=sub[j]);
    i--;
    while (source[i]&& sub[j] && source[i++] == sub[j++] );
    if (!sub[j])
    {
      result++;
    }
  }
  return result;
}

int main()
{
  char *  str = "";
  char *sub = "fuck";
  char res[50] ="";

  str = "12fuck345";
  delete_sub_str(str, sub, res);
  cout<<res<<endl;

  str = "12fuck";
  delete_sub_str(str, sub, res);
  cout<<res<<endl;


  str = "12fuck345fuck678fuck9";
  cout<<delete_sub_str(str, sub, res)<<",";
  cout<<res<<endl;

  char str1[] = "12fuck";
  delete_sub_str(str1, sub);
  cout<<str1<<endl;

  char str2[] = "12fuck345";
  delete_sub_str(str2, sub);
  cout<<str2<<endl;

  char str3[] = "12fuck345fuck678fuck9";
  cout<<delete_sub_str(str3, sub)<<",";
  cout<<str3<<endl;

  char str4[] = "12fuck345fuck678fuck9fuc";
  cout<<delete_sub_str(str4, sub)<<",";
  cout<<str4<<endl;

  cout<<count_sub("12fuck345fuck678fuck9fuc","fuck")<<endl;
  /*
     12345
     12
     3,123456789
     12
     12345
     3,123456789
     3,123456789fuc
     3
     */
  return 0;
}

File: src/delete_sub_str.java
public class delete_sub_str {
	static int count_sub(char[] source, char[] sub) {
		int count = 0;
		int i = 0;
		int j = 0;
		int nav = 0;

		while (i < source.length) {
			nav = i;
			j = 0;
			while (nav < source.length && j < sub.length
					&& source[nav++] == sub[j++])
				;
			if (j == sub.length) {
				count++;
				i = nav;
			}
			i++;
		}
		return count;
	}

	static char[] delete_sub_str_method(char[] str, char[] sub_str) {
		char[] result_str = new char[str.length - sub_str.length];
		int i = 0;
		int j = 0;
		int index = 0, nav = 0;

		while (i < str.length) {
			nav = i;
			j = 0;
			while (nav < str.length && j < sub_str.length
					&& str[nav++] == sub_str[j++])
				;
			if (j == sub_str.length) {
				i = nav;
			} else {
				while (i < nav) {
					result_str[index++] = str[i++];
				}
			}
			while (i < str.length && str[i] != sub_str[0])
				result_str[index++] = str[i++];
		}
		return result_str;
	}

	public static void main(String arg[]) {
		char[] sub = "fuck".toCharArray();

		System.out
				.println(delete_sub_str_method("12fuck345".toCharArray(), sub));

		System.out.println(delete_sub_str_method("12fuck".toCharArray(), sub));

		System.out.println((delete_sub_str_method(
				"12fuck345fuck678fuck9".toCharArray(), sub)));

		System.out.println(delete_sub_str_method("12fuck".toCharArray(), sub));

		System.out
				.println(delete_sub_str_method("12fuck345".toCharArray(), sub));

		System.out.println(delete_sub_str_method(
				"12fuck345fuck678fuck9".toCharArray(), sub));

		System.out.println(delete_sub_str_method(
				"12fuck345fuck678fuck9fuc".toCharArray(), sub));

		System.out.println(delete_sub_str_method(
				"12fck345fuck678fuck9fuc".toCharArray(), sub));

		System.out.println(count_sub("12fuck345fuck678fuck9fuc".toCharArray(),
				"fuck".toCharArray()));
		System.out.println(count_sub("12fuk345fuck678fuck9fuc".toCharArray(),
				"fuck".toCharArray()));
		/*
		 * 12345 12 3,123456789 12 12345 3,123456789 3,123456789fuc 3
		 */

	}
}
